src/data_vis.ts

```diff
@@ -183,7 +183,7 @@ class DataVisModel {
 
     updateXAxisOptions() {
         // refilter the data
-        let chartOptions = this.chartOptions();
+        let chartOptions = { ...this.chartOptions() };
```

Comment:
This is actually a very idiomatic way of doing things, you can also combine lines 186/187 by doing
``` const chartOptions = {...this.chartOptions(), maxPlot: -1}; ```